undertow/src/main/java/org/wildfly/extension/undertow/HttpUnavailablePathHandler.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2015, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.wildfly.extension.undertow;
+
+import io.undertow.server.HttpHandler;
+import io.undertow.server.HttpServerExchange;
+import io.undertow.server.handlers.PathHandler;
+import io.undertow.util.PathMatcher;
+
+import org.jboss.logging.Logger;
+
+/**
+ * Handler which keeps track of registered paths. If it is enabled, it will make HttpServerExchange return configured error
+ * code. Otherwise it will trigger PathHandler.
+ *
+ * @author baranowb
+ *
+ */
+public class HttpUnavailablePathHandler extends PathHandler {
+
+    protected static final Logger log = Logger.getLogger(HttpUnavailablePathHandler.class);
+    protected static final boolean traceEnabled;
+    static {
+        traceEnabled = log.isTraceEnabled();
+    }
+
+    private CustomResponseCodeHandler customeResponseHandler = null;
+
+    public HttpUnavailablePathHandler(final HttpHandler defaultHandler) {
+        super(defaultHandler);
+    }
+
+    public HttpUnavailablePathHandler() {
+    }
+    @Override
+    public void handleRequest(HttpServerExchange exchange) throws Exception {
+        if(this.customeResponseHandler == null){
+            super.handleRequest(exchange);
+            return;
+        }
+
+        final PathMatcher.PathMatch<HttpHandler> match = super.pathMatcher.match(exchange.getRelativePath());
+        if(match.getRemaining().equals(exchange.getRelativePath())){
```

Comment:
Than it looks a bit like a bug. If there is a match, getRemaining( ) should return 'null' or ""